lighthouse-core/audits/valid-source-maps.js

```diff
@@ -0,0 +1,158 @@
+      requiredArtifacts: ['ScriptElements', 'SourceMaps', 'URL'],
+    };
+  }
+
+  /**
+   * Returns true if the size of the script exceeds a static threshold.
+   * @param {LH.Artifacts.ScriptElement} scriptElement
+   * @param {string} finalURL
+   * @return {boolean}
+   */
+  static isLargeFirstPartyJS(scriptElement, finalURL) {
+    if (scriptElement.content === null) return false;
+
+    const isLargeJS = scriptElement.content.length >= LARGE_JS_BYTE_THRESHOLD;
+    const isFirstPartyJS = scriptElement.src ?
+      thirdPartyWeb.isFirstParty(scriptElement.src, thirdPartyWeb.getEntity(finalURL)) : false;
+
+    return isLargeJS && isFirstPartyJS;
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   */
+  static async audit(artifacts) {
+    const {SourceMaps} = artifacts;
+
+    /** @type {Set<string>} */
+    const isMissingMapForLargeFirstPartyScriptUrl = new Set();
+
+    let missingMapsForLargeFirstPartyFile = false;
+    const results = [];
+    for (const scriptElement of artifacts.ScriptElements) {
+      if (!scriptElement.src) continue; // TODO: inline scripts, how do they work?
+
+      const sourceMap = SourceMaps.find(m => m.scriptUrl === scriptElement.src);
+      const errors = [];
+      const isLargeFirstParty = this.isLargeFirstPartyJS(scriptElement, artifacts.URL.finalUrl);
+
+      if (isLargeFirstParty && (!sourceMap || !sourceMap.map)) {
+        missingMapsForLargeFirstPartyFile = true;
+        isMissingMapForLargeFirstPartyScriptUrl.add(scriptElement.src);
+        errors.push({error: str_(UIStrings.missingSourceMapErrorMessage)});
+      }
+
+      if (sourceMap && !sourceMap.map) {
+        errors.push({error: sourceMap.errorMessage});
+      }
+
+      // Sources content errors.
+      if (sourceMap && sourceMap.map) {
+        const sourcesContent = sourceMap.map.sourcesContent || [];
+        let missingSourcesContentCount = 0;
+        for (let i = 0; i < sourceMap.map.sources.length; i++) {
+          if (sourcesContent.length < i || !sourcesContent[i]) missingSourcesContentCount += 1;
+        }
+        if (missingSourcesContentCount > 0) {
+          errors.push({error: str_(UIStrings.missingSourceMapItemsWarningMesssage,
+              {missingItems: missingSourcesContentCount})});
+        }
+      }
+
+      if (sourceMap || errors.length) {
+        results.push({
+          scriptUrl: scriptElement.src,
+          sourceMapUrl: sourceMap && sourceMap.sourceMapUrl,
+          subItems: /** @type {LH.Audit.Details.TableSubItems} */ ({
+            type: 'subitems',
+            items: errors,
+          }),
+        });
+      }
+    }
+
+    /** @type {LH.Audit.Details.TableColumnHeading[]} */
+    const headings = [
+      /* eslint-disable max-len */
+      {
+        key: 'scriptUrl',
+        itemType: 'url',
+        subItemsHeading: {key: 'error'},
+        text: str_(i18n.UIStrings.columnURL),
+      },
+      {key: 'sourceMapUrl', itemType: 'url', text: 'Map URL'}, // TODO uistring
```

Comment:
i18n

Reply:
actually ... maybe this is something not to be translated ... @exterkamp  ?

Reply:
The text "Map URL"? 

I think it should be a translatable string. This isn't a literal that should be preserved, so it should be translated.

This will require a pretty high level of comments to explain what a "map" is in this case.